pwnlib/util/cyclic.py

```diff
@@ -129,19 +129,10 @@ def cyclic(length = None, alphabet = None, n = None):
         log.error("Can't create a pattern length=%i with len(alphabet)==%i and n==%i" \
                   % (length, len(alphabet), n))
 
-    out = []
-    for ndx, c in enumerate(de_bruijn(alphabet, n)):
-        if length != None and ndx >= length:
-            break
-        else:
-            out.append(c)
+    generator = de_bruijn(alphabet, n)
+    out = [next(generator) for _ in range(length)] if length != None else list(generator)
```

Comment:
You could also use [iters.take](/Gallopsled/pwntools/blob/2116bf3118612a5f53a9a02a8d46ed2f8b1cefcb/pwnlib/util/iters.py#L72) here I guess, as it works also with `n=None`.

Reply:
Will do.